operator/src/main/java/org/keycloak/operator/KeycloakDeployment.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.keycloak.operator.crds.Keycloak;
+import org.keycloak.operator.crds.KeycloakSpec;
+import org.keycloak.operator.crds.KeycloakStatus;
+
+import static org.keycloak.operator.crds.KeycloakStatus.State.*;
+
+public class KeycloakDeployment {
+
+    KubernetesClient client = null;
+
+    KeycloakDeployment(KubernetesClient client) {
+        this.client = client;
+    }
+
+    public Deployment getKeycloakDeployment(Keycloak keycloak) {
+        // TODO this should be done through an informer to leverage caches
+        // WORKAROUND for: https://github.com/java-operator-sdk/java-operator-sdk/issues/781
+        return client
+                .apps()
+                .deployments()
+                .inNamespace(keycloak.getMetadata().getNamespace())
+                .list()
+                .getItems()
+                .stream()
+                .filter((d) -> d.getMetadata().getName().equals(org.keycloak.operator.Constants.NAME))
+                .findFirst()
+                .orElse(null);
+//                .withName(Constants.NAME)
+//                .get();
+    }
+
+    public void createKeycloakDeployment(Keycloak keycloak) {
+        client
+            .apps()
+            .deployments()
+            .inNamespace(keycloak.getMetadata().getNamespace())
+            .create(newKeycloakDeployment(keycloak));
+    }
+
+    public Deployment newKeycloakDeployment(Keycloak keycloak) {
+        return new DeploymentBuilder()
```

Comment:
Just something to discuss for a follow-up task (doesn't block this PR). Maybe we could use a YAML file template and just modify the variable parts in it instead of doing this spaghetti. :) WDYT?

Reply:
I think that this needs to stay in code to give us the flexibility of tuning and merging different values .... maybe with a far better formatting though ...

Reply:
With sort of "template" we would not loose flexibility. We'd just get rid of the static stuff in the code and keep only the things that needs to be dynamic, making the code much more readable. But as I said, we can improve this in the future.

Reply:
I thought twice at this and I'm happy to give it a spin!

Reply:
Done and I like it, great suggestion @vmuzikar !

Reply:
@andreaTP Thanks for the update, looks great now. ;)
Could you please mark this as resolved?